### DISCUSSION TITLE
How to manage several python  project with console_scripts in PATH?

### DISCUSSION THREAD

**Discussion by byumov** [pypa/setuptools#3493] [Q&A] [answered]
Hi,
I have a problem with manage several projects with `console_scripts`. We need to put several project to system PATH. 
Let it be `vault-tool`, `config-tool` and `other-tool`.

All three *-tool is a separated projects with their own repo, setup.py and development cycle.
Each project installed to their virtualenv like this:

```
/opt/python-vault/bin/python
/opt/python-config/bin/python
/opt/python-other/bin/python
```

We need all scripts in `PATH`, so:
`PATH=$PATH:/opt/python-vault/bin:/opt/python-config/bin:/opt/python-other/bin`
And there we have a problem, because all three tool depend on `other-tool`. And we have three `other-tool` script (in all virtualenvs). It's a mess.

How I can prevent install console_scripts when package installed as dependency?
Is there a way to put `console_scripts` to other directory (not `./bin/`)?
Any other ideas? Thanks.

**Comment by abravalheri** [comment, marked as answer]
> How I can prevent install console_scripts when package installed as dependency?

Hi @byumov, I don't know about any tool in that allows you not installing `console_scripts`. This is probably a question that you might want to direct to a installer, like `pip`, since `setuptools` is trying to reduce its scope and avoid performing the installation of packages itself (`python setup.py install` is considered deprecated).

> Is there a way to put `console_scripts` to other directory (not ./bin/)?

I am afraid that this is also outside of the scope of the `setuptools` project. Maybe you can try to play with [`installer`](https://installer.readthedocs.io/en/stable/)? However, you probably will need to write your own script and handle dependencies directly (as the existing `installer` CLI does not allow specifying the paths).

---

Instead if adding the separated virtual environments' `bin` folder to `PATH`, how about you create a symbolic link to the executables (e.g. `vault-tool`, `config-tool` and `other-tool`) inside a different folder (that is already on `$PATH`, e.g. `$HOME/.local/bin`, or that can be added to path)?

Alternatively you might also consider using something like https://pypa.github.io/pipx/.
